src/bootstrap/src/ferrocene/dist.rs

```diff
@@ -431,3 +405,39 @@ impl Step for GenerateBuildMetadata {
         builder.copy_to_folder("ferrocene/packages.toml".as_ref(), dist_dir.as_ref());
     }
 }
+
+impl GenerateBuildMetadata {
+    pub(super) fn channel(
```

Comment:
Nit: I'd move this to a standalone function `ferrocene_channel` in `src/bootstrap/src/ferrocene/mod.rs`